src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -295,8 +304,11 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     stateParams.caseSessionManager = Platform::New<CASESessionManager>();
     ReturnErrorOnFailure(stateParams.caseSessionManager->Init(stateParams.systemLayer, sessionManagerConfig));
 
-    ReturnErrorOnFailure(chip::app::InteractionModelEngine::GetInstance()->Init(
-        stateParams.exchangeMgr, stateParams.fabricTable, stateParams.reportScheduler, stateParams.caseSessionManager));
+    chip::app::InteractionModelEngine * interactionModelEngine = chip::app::InteractionModelEngine::GetInstance();
+
+    interactionModelEngine->SetDataModelProvider(params.dataModelProvider);
```

Comment:
This looks wrong.  InitDataModelHandler() happened above, and our ordering guarantee is that SetDataModelProvider() happens before that, no?  We should match the Server ordering of these things for now (and make sure to update the orders in sync).

That said, it's really weird that things are using the interaction model engine (to get the data model provider) before we have inited it.  We really need to think about the right way to keep track of these sorts of "global stack state".  But followup for that, obviously.

Reply:
Moved it higher above. I was relying on integration and unit tests to catch things and those passed even with a null data model (still fixing some unit tests).